# Coding Discussion 02
### Ella Zhang

## Preparation

### Read in the data.

In [1]:
import csv
with open('us-states.csv') as file:
        state_covid_data = []
        for row in csv.reader(file):
            state_covid_data.append(row)

len(state_covid_data)

10080

### View the first 5 rows of the data.

In [2]:
state_covid_data[:5]

[['date', 'state', 'fips', 'cases', 'deaths'],
 ['2020-01-21', 'Washington', '53', '1', '0'],
 ['2020-01-22', 'Washington', '53', '1', '0'],
 ['2020-01-23', 'Washington', '53', '1', '0'],
 ['2020-01-24', 'Illinois', '17', '1', '0']]

## Question

### (1) Count up the number of _unique_ dates in the data. 

In [3]:
unique_dates = set() # Create an empty set to store unique dates
for row in state_covid_data[1:]:
    unique_dates.add(row[0])

print(len(unique_dates))

225


There are 225 unique dates in the data.

### (2) Find the first date in which the District of Columbia recorded a case. 

In [4]:
for row in state_covid_data[1:]:
    if row[1] == "District of Columbia":
        print(row[0])
        break

2020-03-07


The first case of COVID-19 in the District of Columbia was recorded on March 7, 2020.

### (3) Write a function that takes in a _state name_ as input (e.g. "Wisconsin") and outputs the date of its first case.

In [5]:
def first_date(state_name):
    """
    This is a function that takes in a state name as input and outputs the date of its first case.

    Args:
        staten_name (str): state name listed in state_covid_data

    Returns:
        str: date of the first case in the input state

    """
    for row in state_covid_data[1:]:
        if row[1] == state_name:
            firstdate = row[0]
            break
    return firstdate

first_date("Wisconsin")

'2020-02-05'

### (Optional) Bonus
### Write a function that takes in a _state name_ as input (e.g. "Wisconsin") and outputs the date when the number of reported cases within the state exceeded 1000.

In [6]:
def date_exceeded_1000(state_name):
    """
    This is a function that takes in a state name as input and outputs the date when the number of reported cases within the state exceeded 1000.

    Args:
        state_name (str): state name listed in state_covid_data

    Returns:
        str: date when the number of reported cases within the input state exceeded 1000

    """

    for row in state_covid_data[1:]:
        cases = int(row[3])
        if row[1] == state_name and cases >= 1000:
                break
    return row[0]

date_exceeded_1000("Wisconsin")

'2020-03-28'